### DISCUSSION TITLE
Support of TOML format in addition (or in replace) to INI file format

### DISCUSSION THREAD

**Discussion by bagustris** [felixbur/nkululeko#177] [Ideas]
Starting from Python 3.11, tomlib and toml are shipped in standard library, meaning no need to install additional packages to read .toml file.

I found this info when reading a manual [`configparser`](https://docs.python.org/3/library/configparser.html) (in See also:). Based on its explanation, in [https://toml.io/en/](https://toml.io/en/), TOML format has many advantages over INI format, e.g., standardization, can read different values automatically: integer, boolean, strings (while INI only can read string only, afaik). I think this can reduce the burden of future contributor due to the large use of `ast.literal_eval()` to parse value in INI format.

PROS:
- standardize
- flexibility when reading configuration, meaning we can support more configuration for Nkululeko e.g., number of layers, etc. 

CONS:  
- minimum Python is 3.11 (current minimum is 3.9)
- need additional effort

As a backup, we can support both formats (TOML and INI) in v1.0, while give depreciation warning in the next major version (v2.0?)

**Comment by felixbur** [comment]
sounds good, but I'd rather only support one format (actually in my new projects i use hydra).
So i suggest we wait a while until 3.11 is more "normal" and then switch.
